pkg/migration/azure_crs.go

```diff
@@ -0,0 +1,94 @@
+package migration
+
+import (
+	"context"
+	"fmt"
+
+	provider "github.com/giantswarm/apiextensions/v3/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/microerror"
+	capz "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	capzexp "sigs.k8s.io/cluster-api-provider-azure/exp/api/v1alpha3"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	capiexp "sigs.k8s.io/cluster-api/exp/api/v1alpha3"
+	ctrl "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (m *azureMigrator) readAzureConfig(ctx context.Context) error {
+	objList := &provider.AzureConfigList{}
+	selector := ctrl.MatchingLabels{capi.ClusterLabelName: m.clusterID}
+	err := m.mcCtrlClient.List(ctx, objList, selector)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if len(objList.Items) == 0 {
+		return microerror.Mask(fmt.Errorf("AzureConfig not found for %q", m.clusterID))
+	}
+
+	obj := objList.Items[0]
+	m.crs[obj.Kind] = &obj
+
+	return nil
+}
+
+func (m *azureMigrator) readCluster(ctx context.Context) error {
+	objList := &capi.ClusterList{}
+	selector := ctrl.MatchingLabels{capi.ClusterLabelName: m.clusterID}
+	err := m.mcCtrlClient.List(ctx, objList, selector)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if len(objList.Items) == 0 {
```

Comment:
Since we are filtering by label, does it make sense to check for `!= 1` instead? I can imagine cases where we have 2 cluster CRs with the same cluster id label and troubleshooting could become problematic

Reply:
I'll add separate case for `len() > 1` so the error message is more explicit.